tool/tctl/common/loadtest_command.go

```diff
@@ -0,0 +1,148 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/google/uuid"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+)
+
+// LoadtestCommand implements the `tctl loadtest` family of commands.
+type LoadtestCommand struct {
+	config *servicecfg.Config
+
+	nodeHeartbeats *kingpin.CmdClause
+
+	count       int
+	interval    time.Duration
+	ttl         time.Duration
+	concurrency int
+}
+
+// Initialize allows LoadtestCommand to plug itself into the CLI parser
+func (c *LoadtestCommand) Initialize(app *kingpin.Application, config *servicecfg.Config) {
+	c.config = config
+	loadtest := app.Command("loadtest", "Tools for generating artificial load").Hidden()
+
+	c.nodeHeartbeats = loadtest.Command("node-heartbeats", "Generate artifical node heartbeats").Hidden()
+	c.nodeHeartbeats.Flag("count", "Number of unique nodes to heartbeat").Default("10000").IntVar(&c.count)
+	c.nodeHeartbeats.Flag("interval", "Node heartbeat interval").Default("1m").DurationVar(&c.interval)
+	c.nodeHeartbeats.Flag("ttl", "TTL of heartbeated nodes").Default("10m").DurationVar(&c.ttl)
+	c.nodeHeartbeats.Flag("concurrency", "Max concurrent requests").Default(
+		strconv.Itoa(runtime.NumCPU() * 16),
+	).IntVar(&c.concurrency)
+}
+
+// TryRun takes the CLI command as an argument (like "loadtest node-heartbeats") and executes it.
+func (c *LoadtestCommand) TryRun(ctx context.Context, cmd string, client auth.ClientI) (match bool, err error) {
+	switch cmd {
+	case c.nodeHeartbeats.FullCommand():
+		err = c.NodeHeartbeats(ctx, client)
+	default:
+		return false, nil
+	}
+	return true, trace.Wrap(err)
+}
+
+func (c *LoadtestCommand) NodeHeartbeats(ctx context.Context, client auth.ClientI) error {
+	infof := func(format string, args ...any) {
+		fmt.Fprintf(os.Stderr, "[i] "+format+"\n", args...)
+	}
+	warnf := func(format string, args ...any) {
+		fmt.Fprintf(os.Stderr, "[!] "+format+"\n", args...)
+	}
+
+	infof("Setting up node hb load generation. count=%d, interval=%s, ttl=%s, concurrency=%d",
+		c.count,
+		c.interval,
+		c.ttl,
+		c.concurrency,
+	)
+
+	node_ids := make([]string, 0, c.count)
+
+	for i := 0; i < c.count; i++ {
+		node_ids = append(node_ids, uuid.New().String())
+	}
+
+	workch := make(chan string, c.count*2)
+
+	errct := new(atomic.Uint64)
```

Comment:
Unless it tries to copy the value in the background goroutines:

```suggestion
	var errct atomic.Uint64
```